rclcpp/test/rclcpp/test_time.cpp

```diff
@@ -848,3 +849,64 @@ TEST_F(TestClockSleep, sleep_for_basic_ros) {
   sleep_thread.join();
   EXPECT_TRUE(sleep_succeeded);
 }
+
+class TestClockValid : public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    rclcpp::shutdown();
```

Comment:
I'm surprised this is needed in setup since teardown calls it. What happens without it?

Reply:
Ah, I was mimicking https://github.com/methylDragon/rclcpp/blob/ch3/clock-wait-for-valid/rclcpp/test/rclcpp/test_time.cpp#L461 

I don't think anything goes wrong if it gets removed though, but I put it in to be safe

Reply:
Removed for my new test case, it seems necessary for the other one, for some reason..

https://github.com/ros2/rclcpp/compare/937c2130e827a6216389e68a0ab9c08d3f3b5e62..23271a6bd124f9345b86a82441810588d35b5de6